packages/expect/src/asymmetric_matchers.js

```diff
@@ -218,6 +222,10 @@ class StringMatching extends AsymmetricMatcher {
   }
 
   asymmetricMatch(other: string) {
+    if (!isA('String', other)) {
```

Comment:
According to [our docs](http://facebook.github.io/jest/docs/en/expect.html#expectstringmatchingregexp), `stringMatching()` accepts a regexp. Mind changing the type and runtime check?

Reply:
cc @rogeliog can you take a look at this?

Reply:
@thymikee The `StringMatching` constructor does indeed take either string or RegExp, but argument of the `asymmetricMatch` function takes only string as received value, true?

That is, `expect.stringMatching(/^Alic/)` returns `true` when the received value is `'Alicia'` or `false` when the received value is `'Bob'` but should report an error if the received value is a RegExp, because how to decide if a **received** RegExp matches an **expected** RegExp?

Reply:
@pedrottimark you're right, I mistaken `other` with `sample`